layer/nrlf/core/dynamodb/model.py

```diff
@@ -69,30 +87,60 @@ def from_document_reference(
         cls, resource: DocumentReference, created_on: Optional[str] = None
     ) -> "DocumentPointer":
         resource_id = getattr(resource, "id")
-        coding = getattr(resource.type, "coding")
+
+        # Get identifiers
         subject_identifier = getattr(resource.subject, "identifier")
         custodian_identifier = getattr(resource.custodian, "identifier")
+        if not resource.author or len(resource.author) != 1:
+            raise ValueError("DocumentReference.author must have exactly one item")
+        author_identifier = getattr(resource.author[0], "identifier")
 
+        # Get identifier values
         nhs_number = getattr(subject_identifier, "value")
         custodian = getattr(custodian_identifier, "value")
+        author = getattr(author_identifier, "value")
+        master_identifier = getattr(resource.masterIdentifier, "value", None)
```

Comment:
The system of the master_identifier field can be set to anything. We're only going to use the value for the GSI, producers will be able to search by master_identifier and they'll get all pointers they have created.